pytorch_lightning/utilities/cli.py

```diff
@@ -89,20 +89,24 @@ def __init__(
         parser: LightningArgumentParser,
         config: Union[Namespace, Dict[str, Any]],
         config_filename: str,
+        exist_ok: bool = False,
     ) -> None:
         self.parser = parser
         self.config = config
         self.config_filename = config_filename
+        self.exist_ok = exist_ok
 
     def on_train_start(self, trainer: Trainer, pl_module: LightningModule) -> None:
         log_dir = trainer.log_dir or trainer.default_root_dir
         config_path = os.path.join(log_dir, self.config_filename)
-        if os.path.isfile(config_path):
+        if not self.exist_ok and os.path.isfile(config_path):
             raise RuntimeError(
-                f'{self.__class__.__name__} expected {config_path} to not exist. '
-                'Aborting to avoid overwriting results of a previous run.'
+                f'{self.__class__.__name__} expected {config_path} to NOT exist. Aborting to avoid overwriting'
+                ' results of a previous run. You can delete the previous config file,'
+                ' set `LightningCLI(save_config_callback=None)` to disable config saving,'
+                ' or set `LightningCLI(save_config_exist_ok=True)` to overwrite the config file.'
```

Comment:
I would go for `save_config_overwrite` or similar because if we think about `makedirs exist_ok` it does not overwrite the folder.